### PR TITLE
docs: Revamp root and nested AGENTS.md

### LINKED ISSUE #6638: Revamp AGENTS.md
Check AGENTS.md and revamp if needed.

Some useful resources and notes:

[https://github.blog/ai-and-ml/github-copilot/how-to-write-a-great-agents-md-lessons-from-over-2500-repositories/](<https://github.blog/ai-and-ml/github-copilot/how-to-write-a-great-agents-md-lessons-from-over-2500-repositories/>)

solid advice imho if you’re adding/refreshing `AGENTS.md` for a SDK

* specialist > vague “helpful assistant”
* real commands (with exact flags) early
* one code example > three paragraphs of prose
* explicit boundaries: always / ask first / never
* naming the stack precisely (versions + key deps)
* cover the boring stuff: commands, testing, structure, style, git, boundaries

worth a look at how we do it for junior: same stuff, different shape:

* shared repo `AGENTS.md` as the operating manual (commands, testing, architecture)
* skills as the specialists (loaded on demand), not persona files (as in the blog)

https://github.com/getsentry/junior/blob/main/AGENTS.md
https://github.com/getsentry/junior/blob/main/policies/agent-steering.md

### PR DESCRIPTION
**Pull request by antonis**
## :loudspeaker: Type of change
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring

## :scroll: Description
Revamps the agent operating manual (`AGENTS.md`, symlinked as `CLAUDE.md`) and **all** nested `AGENTS.md` files against the "how to write a great AGENTS.md" checklist.

Root `AGENTS.md`:

- **`## Boundaries`** — a three-tier ✅ Always / ⚠️ Ask first / 🚫 Never section encoding the decisions an agent can't derive from the code (arch + platform parity, PII gating, bridge/ABI deprecation path, generated-file edits, CHANGELOG placement).
- **`## Skills — load on demand`** — an index mapping each local skill (`spec`, `design-first`, `code-guidelines`, `test-guidelines`, `review`, `diagnosing-bugs`) to when to reach for it, plus a note on the remote specialists pulled via `agents.toml`.
- **`## Stack`** — reframes the intro as a hybrid SDK (TS/JS wrapping sentry-cocoa + sentry-android/java via the bridge) and, rather than copying rot-prone version strings, points at the live sources: `packages/core/package.json` for JS/peer versions and the release-generated `SDK-VERSIONS.md` for bundled native versions.
- Corrected the generated-files boundary (dropped a stale `.yalc` reference) and refined the CHANGELOG rule so notable internal changes land under `### Internal` rather than being excluded outright.

`packages/core/AGENTS.md`:

- Added a skill-precedence pointer ("depth lives in the skills; where a convention here overlaps a skill, the skill wins").
- Fixed drifted diagnostics examples that used `logger` (the Logs API — recurses in the bridge hot path) to `debug`, matching the SDK's own source (52 files import `{ debug }` from `@sentry/core`) and the file's own TurboModule warning.

Nested surface + sample files:

- **`packages/core/android/AGENTS.md`** & **`packages/core/ios/AGENTS.md`** — added the skill-precedence pointer and a scoped Boundaries block (each tailored to its surface), and fixed drifted bridge examples: the real reject error code is the shared `SentryReactNative`, not a per-method `OPERATION_FAILED`; dropped the iOS `[SentryLog logWithMessage:]` snippet that appears nowhere in the ObjC layer.
  - _Android boundaries:_ land a bridge change in **both** `src/oldarch/` + `src/newarch/`, catch-and-`reject` at every `@ReactMethod`, gate PII, keep the codegen ABI mirrored/backward-compatible, and don’t hand-bump `io.sentry:sentry-android` (use `scripts/update-android.sh`).
  - _iOS boundaries:_ always resolve-or-reject (an unfinished `Promise` hangs the JS caller), catch native exceptions, gate PII, route hybrid-SDK access through `RNSentryInternal` instead of `PrivateSentrySDKOnly`, and don’t hand-bump sentry-cocoa (use `scripts/update-cocoa.sh`).
- **`samples/react-native/AGENTS.md`** — documented the iOS `yarn pod-install-<debug|release>-<static|dynamic>[-legacy]` build matrix (each script sets `ENABLE_PROD` / `USE_FRAMEWORKS` / `ENABLE_NEW_ARCH`) and the Android `newArchEnabled` toggle, so a native change gets verified across the combinations it affects.
- **`samples/expo/AGENTS.md`** — added the real `run:ios` / `run:android` scripts and the dev-client caveat (native code means plain Expo Go isn't enough).

## :bulb: Motivation and Context
Closes #6638. The always-loaded manual was missing the highest-value guidance (boundaries, when to load which skill) and had drifted from the code in places. Version numbers are deliberately not copied into prose — they point at existing sources so they can't rot into confident falsehoods in a file agents treat as ground truth.

## :green_heart: How did you test it?
Docs-only change. Every factual claim was verified against the codebase: package/peer/native versions against `package.json`/`RNSentry.podspec`/`build.gradle`; all referenced skills against `agents.toml`; every command against `package.json` scripts; the `SentryReactNative` reject code and `NSLog`/`logger.log` idioms against the actual ObjC/Java source; the android arch dirs, `RNSentryInternal.swift`, `scripts/update-{cocoa,android}.sh`, and the sample arch toggles against the tree.

## :pencil: Checklist
- [ ] I added tests to verify changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] I updated the wizard if needed.
- [ ] All tests passing.
- [x] Public API changes reviewed by another Mobile SDK team member or implemented according to the [develop docs](https://develop.sentry.dev/) spec.
- [x] No breaking changes.

## :crystal_ball: Next steps
None — the revamp now covers the root manual plus every nested `AGENTS.md` (`packages/core`, `android`, `ios`, `samples/react-native`, `samples/expo`).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by antonis** [comment]
@sentry review

File: packages/core/android/AGENTS.md [hunk 1]
```diff
@@ -25,25 +27,39 @@
...
+
 ```java
 @ReactMethod
 public void nativeOperation(String param, Promise promise) {
   try {
     boolean result = performOperation(param);
     promise.resolve(result);
   } catch (Exception e) {
-    promise.reject("OPERATION_FAILED", "Operation failed: " + e.getMessage(), e);
+    promise.reject("SentryReactNative", e.getMessage(), e);
```
[the remark this answers is not in the document]
Comment (antonis) [reply packages/core/android/AGENTS.md:47]: Good catch — fixed in 57f8da29. Updated the example (and the Boundaries bullet) to `catch (Throwable e)`, matching the boundary rule and `RNSentryModuleImpl`, which catches `Throwable` throughout so an `Error` can't crash the host app.

**Review by lucas-zimerman — APPROVED** [review]
LGTM!
